pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -51,18 +57,76 @@ var ManifestPath = "./bindata"
 // Add creates a new OperConfig Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager, status *statusmanager.StatusManager, c cnoclient.Client) error {
-	return add(mgr, newReconciler(mgr, status, c))
+	rc, err := newReconciler(mgr, status, c)
+	if err != nil {
+		return err
+	}
+	return add(mgr, rc)
 }
 
 const ControllerName = "operconfig"
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager, status *statusmanager.StatusManager, c cnoclient.Client) *ReconcileOperConfig {
-	return &ReconcileOperConfig{
-		client: c,
-		status: status,
-		mapper: mgr.GetRESTMapper(),
+func newReconciler(mgr manager.Manager, status *statusmanager.StatusManager, c cnoclient.Client) (*ReconcileOperConfig, error) {
+	ctx := context.Background()
+	ctx, cancelFn := context.WithCancel(ctx)
+	stopCh := make(chan struct{})
+	go func() {
+		defer cancelFn()
+		<-stopCh
+	}()
+	kubeConfig := c.Default().Config()
+	kubeClient := c.Default().Kubernetes()
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+	configInformers := configinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	desiredVersion := os.Getenv("RELEASE_VERSION")
+	missingVersion := "0.0.1-snapshot"
+
+	eventRecorder := events.NewKubeRecorder(kubeClient.CoreV1().Events("openshift-network-operator"), "cluster-network-operator", &corev1.ObjectReference{
+		APIVersion: "apps/v1",
+		Kind:       "Deployment",
+		Namespace:  "openshift-network-operator",
+		Name:       "network-operator",
+	})
+
+	// By default, this will exit(0) the process if the featuregates ever change to a different set of values.
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(), configInformers.Config().V1().FeatureGates(),
+		eventRecorder,
+	)
+	go featureGateAccessor.Run(ctx)
+	go configInformers.Start(stopCh)
+
+	select {
+	case <-featureGateAccessor.InitialFeatureGatesObserved():
+		featureGates, _ := featureGateAccessor.CurrentFeatureGates()
+		if err != nil {
+			return nil, err
+		} else {
+			klog.Infof("FeatureGates initialized: knownFeatureGates=%v", featureGates.KnownFeatures())
+		}
+	case <-time.After(1 * time.Minute):
+		return nil, fmt.Errorf("timed out waiting for FeatureGate detection")
 	}
+
+	featureGates, err := featureGateAccessor.CurrentFeatureGates()
+	if err != nil {
+		return nil, err
+	}
+	// example of future featuregate read and usage to set a variable to pass to a controller
+	featureGatesCNO := make(map[string]bool, 0)
```

Comment:
`featureGatesCNO` isn't used anywhere now, we should remove that variable together with `adminNetworkPolicyEnabled`?

Reply:
yes hehe it was a rebase glitch, removed this in my new version.
thanks pat :D